bin/improver-tests

```diff
@@ -18,13 +18,79 @@ function echo_ok {
     echo -e "\033[1;32m[OK]\033[0m $1"
 }
 
+function echo_fail {
+    echo -e "\033[1;31m[FAIL]\033[0m $1"
+}
+
 function get_python_files {
     FILES_TO_TEST=`find $IMPROVER_DIR -type f \( -name '*.py' \
                                               -o -name 'improver-*' \
                                                ! -name 'improver-tests' \
                                                ! -name '*~' \)`
 }
 
+function licence_check {
+    # Iterate through files to check whether they contain the expected
+    # licence information. If any files do not contain the expected licence
+    # information, then this test will fail.
+read -d '' expected <<'__TEXT__' || true
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2018 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+__TEXT__
+
+    FILES_TO_TEST=$@
+
+    count=0
+    for FILE in $FILES_TO_TEST; do
+        # Check whether file has a size greater than zero.
+        if [[ -s $FILE ]]; then
+            file_contents=$(<$FILE)
+            if [[ "$file_contents" != *"$expected"* ]]; then
+                echo "Unexpected licence information: ${FILE#$IMPROVER_DIR}"
+                count=$((count+1))
+            fi
+        fi
+    done
+    if (( $count > 0 )); then
+        echo_fail "IMPROVER licences"
+        exit 1
```

Comment:
This makes the entire testing process stop. All the other tests that fail allow the subsequent tests to continue running so that the user gets a complete summary at the end, this should do the same.